utils/py_es_bulk.py

```diff
@@ -0,0 +1,197 @@
+"""
+Opinionated methods for interfacing with Elasticsearch. We provide two
+such opinions for creating templates (put_template()) and bulk indexing
+(streaming_bulk).
+"""
+
+import sys, os, time, json, errno, logging, math
+
+from random import SystemRandom
+from collections import Counter, deque
+from urllib3 import exceptions as ul_excs
+try:
+    from elasticsearch1 import VERSION as es_VERSION, helpers, exceptions as es_excs
+    _es_logger = "elasticsearch1"
+except ImportError:
+    from elasticsearch import VERSION as es_VERSION, helpers, exceptions as es_excs
+    _es_logger = "elasticsearch"
+
+
+# Use the random number generator provided by the host OS to calculate our
+# random backoff.
+_r = SystemRandom()
+# The maximum amount of time (in seconds) we'll wait before retrying an
+# operation.
+_MAX_SLEEP_TIME = 120
+# Always use "create" operations, as we also ensure each JSON document being
+# indexed has an "_id" field, so we can tell when we are indexing duplicate
+# data.
+_op_type = "create"
```

Comment:
Not sure if this is absolutely needed, I don't see a way to index duplicate data. Theres a difference in indexing between pbench and snafu in that pbench has only one end point that does indexing, so you dont want to index the same tarball but with snafu the data is generated and then indexed, so we should remove this as its an unnecessary computation that we dont hace to do.

Reply:
Please see above response.